libbpf-rs/src/ringbuf.rs

```diff
@@ -146,6 +146,15 @@ impl<'a> RingBuffer<'a> {
 
         util::parse_ret(ret)
     }
+
+    /// Get an fd that can be used to sleep until data is available
+    pub fn epoll_fd(&self) -> Result<()> {
```

Comment:
Shouldn't this be returning an FD?

And does it need a `Result` wrapper? Doesn't the `RingBuffer` hold the invariant that the underlying `struct ring_buffer` is valid?

Reply:
The fd may be an error value if it failed to create the epoll instance (https://elixir.bootlin.com/linux/latest/source/tools/lib/bpf/ringbuf.c#L177)

Reply:
Nvm, just realized that epoll_fd will be called only when fd is value. I'll update this commit :) thanks for catching this!